proof/refine/RISCV64/TcbAcc_R.thy

```diff
@@ -2364,15 +2364,387 @@ lemma in_correct_ready_qD:
    \<Longrightarrow> tcb_domain tcb = d \<and> tcb_priority tcb = p "
   by (fastforce simp: in_correct_ready_q_def vs_all_heap_simps)
 
-lemma ready_or_release'D1:
-  "\<lbrakk>ready_or_release' s'; (tcbInReleaseQueue |< tcbs_of' s') tcbPtr\<rbrakk>
-   \<Longrightarrow> \<not> (tcbQueued |< tcbs_of' s') tcbPtr"
-  by (clarsimp simp: ready_or_release'_def opt_pred_def opt_map_def split: option.splits)
+lemma threadSet_heap_ls_other:
+  "\<lbrace>\<lambda>s. heap_ls (tcbSchedNexts_of s) st ls \<and> t \<notin> set ls\<rbrace>
+   threadSet F t
+   \<lbrace>\<lambda>_ s. heap_ls (tcbSchedNexts_of s) st ls\<rbrace>"
+  apply (wpsimp wp: threadSet_wp)
+  by (fastforce dest: heap_path_heap_upd_not_in)
+
+lemma heap_path_not_Nil:
+  "heap_ls hp (Some st) ls \<Longrightarrow> ls \<noteq> []"
+  by fastforce
+
```

Comment:
This one should go into the `heap_ls` theory.

Reply:
I’ll put up a small PR for this rule for `master` and then once that’s merged, I can cherry pick it onto this branch